integration/tests/params.go

```diff
@@ -0,0 +1,163 @@
+	KubeconfigPath    string
+	KubeconfigTemp    bool
+	TestDirectory     string
+	// privateSSHKeyPath is the SSH key to use for Git operations.
+	PrivateSSHKeyPath       string
+	EksctlCmd               runner.Cmd
+	EksctlCreateCmd         runner.Cmd
+	EksctlGetCmd            runner.Cmd
+	EksctlDeleteCmd         runner.Cmd
+	EksctlDeleteClusterCmd  runner.Cmd
+	EksctlScaleNodeGroupCmd runner.Cmd
+	EksctlUtilsCmd          runner.Cmd
+	EksctlExperimentalCmd   runner.Cmd
+	// Keep track of created clusters, for post-tests clean-up.
+	clustersToDelete []string
+}
+
+// SetRegion sets the provided region and re-generates other fields depending
+// on the region field.
+func (p *Params) SetRegion(region string) {
+	p.Region = region
+	p.GenerateCommands()
+}
+
+// GenerateKubeconfigPath generates a path in ${TEMP} based on these params' cluster name.
+func (p *Params) GenerateKubeconfigPath() {
+	p.KubeconfigPath = path.Join(os.TempDir(), fmt.Sprintf("%s.yaml", p.ClusterName))
+}
+
+// GenerateCommands generates eksctl commands with the various options & values
+// provided to this `Params` object.
+func (p *Params) GenerateCommands() {
+	p.EksctlCmd = runner.NewCmd(p.EksctlPath).
+		WithArgs("--region", p.Region).
+		WithTimeout(30 * time.Minute)
+
+	p.EksctlCreateCmd = p.EksctlCmd.
+		WithArgs("create").
+		WithTimeout(25 * time.Minute)
+
+	p.EksctlGetCmd = p.EksctlCmd.
+		WithArgs("get").
+		WithTimeout(1 * time.Minute)
+
+	p.EksctlDeleteCmd = p.EksctlCmd.
+		WithArgs("delete").
+		WithTimeout(15 * time.Minute)
+
+	p.EksctlDeleteClusterCmd = p.EksctlDeleteCmd.
+		WithArgs("cluster", "--verbose", "4")
+
+	p.EksctlScaleNodeGroupCmd = p.EksctlCmd.
+		WithArgs("scale", "nodegroup", "--verbose", "4").
+		WithTimeout(5 * time.Minute)
+
+	p.EksctlUtilsCmd = p.EksctlCmd.
+		WithArgs("utils").
+		WithTimeout(5 * time.Minute)
+
+	p.EksctlExperimentalCmd = p.EksctlCmd.
+		WithEnv("EKSCTL_EXPERIMENTAL=true")
+}
+
+// NewClusterName generates a new cluster name using the provided prefix, and
+// adds the cluster to the list of clusters to eventually delete, once the test
+// suite has run.
+func (p *Params) NewClusterName(prefix string) string {
+	return p.formatClusterName(prefix, names.ForCluster("", ""))
+}
+
+// FormatClusterName creates a new cluster name using this Params struct's
+// cluster name prefix, the provided name, and adds the cluster to the list of
+// clusters to eventually delete, once the test suite has run.
+func (p *Params) FormatClusterName(name string) string {
+	return p.formatClusterName(p.clusterNamePrefix, name)
+}
+
+func (p *Params) formatClusterName(prefix string, name string) string {
+	clusterName := fmt.Sprintf("it-%s-%s", prefix, name)
+	p.addToDeleteList(clusterName)
+	return clusterName
+}
+
+// addToDeleteList adds the provided cluster name to the list of clusters to eventually delete.
+func (p *Params) addToDeleteList(clusterName string) {
+	p.clustersToDelete = append(p.clustersToDelete, clusterName)
```

Comment:
Even though there's not a lot of contention here, isn't this a potential race condition since `append()` is not thread safe?

Reply:
There could be the day we start to parallelise tests within a test suite, something `ginkgo` doesn't do by default though, so no race at the moment.

Reply:
To be clearer, this is how things are parallelised at the moment:
```
  /- create cluster names array -> run tests -> delete all clusters -\
---- create cluster names array -> run tests -> delete all clusters ---->
  \- create cluster names array -> run tests -> delete all clusters -/
```